app/controllers/works_controller.rb

```diff
@@ -0,0 +1,93 @@
+class WorksController < ApplicationController
+  before_action :find_work, only: [:show, :edit, :update]
+  def index
+    @works = Work.all
+  end
+
+  def show 
+
+    if @work.nil?
+      redirect_to works_path
+      return
+    end
```

Comment:
Since this code is repeated regularly it should also live in `find_work`.  

If you do something like `redirect_to` inside of a controller filter it will immediately return that to the client without running your other code.